semgrep-core/src/engine/Match_search_mode.ml

```diff
@@ -124,7 +124,8 @@ let partition_xpatterns xs =
          match pat with
          | XP.Sem (x, _lang) -> Common.push (x, inside, pid, str) semgrep
          | XP.Spacegrep x -> Common.push (x, pid, str) spacegrep
-         | XP.Regexp x -> Common.push (x, pid, str) regexp
+         | XP.Regexp x ->
+             Common.push (Regexp_engine.pcre_compile x, pid, str) regexp
```

Comment:
you can add a [@@profiling] on Regexp_engine.pcre_compile so we can track (thx to -profile) whether
those calls are expensive.

Reply:
Good point!